src/operations/search.rs

```diff
@@ -1,51 +1,195 @@
+use std::fmt::Display;
+use std::fmt::Formatter;
 use std::str::FromStr;
 
-use crate::internal::commands::ShellCommand;
+use crate::internal::alpm::get_handler;
 use crate::internal::error::SilentUnwrap;
 use crate::internal::exit_code::AppExitCode;
 use crate::internal::rpc::rpcsearch;
+use crate::internal::utils::wrap_text;
+use crate::logging::Printable;
 use crate::Options;
+
 use aur_rpc::SearchField;
+use chrono::Local;
+use chrono::TimeZone;
+use colored::Colorize;
+use trigram::similarity;
+
+#[derive(Debug)]
+pub struct PackageSearchResult {
+    pub repo: String,
+    pub name: String,
+    pub version: String,
+    pub groups: Option<Vec<String>>,
+    pub out_of_date: Option<u64>,
+    pub installed: bool,
+    pub description: String,
+}
+
+impl PackageSearchResult {
+    pub fn score(&self, query: &str) -> f32 {
+        similarity(query, &self.name)
+    }
+}
+
+impl Display for PackageSearchResult {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let repo = &self.repo;
+        let name = &self.name;
+        let version = &self.version;
+        let groups = if self.groups.is_some() {
+            format!(" ({})", self.groups.clone().unwrap().join(", "))
+        } else {
+            "".to_string()
+        };
+        let out_of_date = if self.out_of_date.is_some() {
+            format!(
+                " [out of date: since {}]",
+                Local
+                    .timestamp(self.out_of_date.unwrap().try_into().unwrap(), 0)
```

Comment:
You can use the map stuff to only unwrap once. And instead of unwrapping match inside the if with
`if let Some(...) = self.out_of_date`